### PR TITLE
Tests for react components using enzyme

### PR DESCRIPTION
**Pull request by caalberts**
Enzyme https://github.com/airbnb/enzyme

@ongmin let me know if the hierarchy is correct
cc/ @cbas

### CODE REVIEW

**Comment by cbas** [comment]
@caalberts Looks cool... Doesn't test any events/interativity yet I guess. Also, why have `test` and `test:cover` as two separate ways to run ES6+mocha?

@eddiemoore What do you think of Enzyme to test React components?

**Comment by caalberts** [comment]
@cbas we haven't discussed on the interactivity yet w @ongmin. I intend to write tests for the router too by passing in browser states and asserting the components rendered.

**Comment by eddiemoore** [comment]
@cbas Yeah Enzyme seems pretty good to test components. Although I am just testing with plain React Utils. It also includes a shallow renderer. Enzyme adds some more helpers which seem pretty handy.

**Comment by ongmin** [comment]
@caalberts the address/structure will have to be changed later today. I'll check the ones that are accurate and write a list of the actuals when its clearer.

- [ ] +import Channel from '../app/components/channel'
- [ ] +import Player from '../app/components/player'
- [ ] +import Playlist from '../app/components/playlist'
- [ ] +import Home from '../app/components/homePage'
- [ ] +import ChannelPreview from '../app/components/ChannelPreview'
- [ ] +import Player from '../app/components/Player'
- [ ] +import Playlist from '../app/components/Playlist'
- [ ] +import Thumbnail from '../app/components/Thumbnail'
- [ ] +import Searchbar from '../app/components/Searchbar'

**Comment by cbas** [comment]
@alyssaq FYI

File: package.json [hunk 1]
```diff
@@ -34,17 +39,27 @@
...
     "babel-core": "^6.4.5",
     "babel-eslint": "^4.1.6",
+    "babel-istanbul": "^0.6.0",
     "babel-loader": "^6.2.1",
     "babel-preset-es2015": "^6.3.13",
     "babel-preset-react": "^6.3.13",
     "babel-preset-stage-0": "^6.3.13",
+    "babel-register": "^6.4.3",
+    "chai": "^3.5.0",
+    "codecov.io": "^0.1.6",
```
Comment (eddiemoore) [inline package.json]: @caalberts Note: codecov has updated to a new version just called [codecov](https://www.npmjs.com/package/codecov) which you can just run by using `codecov`. So you don't need to pipe the coverage into the script anymore
CC: @cbas

Comment (cbas) [inline package.json]: :+1: Thanks for the tip!

**Pull request merged**